lib/libc/common/Kconfig

```diff
@@ -1,7 +1,13 @@
 # Copyright (c) 2023 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
+config COMMON_LIBC
+	bool
+	help
+	  library of functions shared across libc variants
+
 config COMMON_LIBC_TIME
 	bool
+	select COMMON_LIBC
 	help
 	  common implementation of time().
```

Comment:
This should be `depends on COMMON_LIBC` since this is a subordinate configuration of `COMMON_LIBC` (the entirety of this file can be wrapped in `if COMMON_LIBC`).

Reply:
I'm not sure that's right, or at least doesn't seem to make sense to me. 

We want to let various C library implementations leverage common implementations of Zephyr-related functionality. Each of those will `select` their necessary pieces. Each of those common libc pieces should `select COMMON_LIBC` so that the library itself will be built whenever the selected C library needs something from it -- that's how this fixes #56243.

Note that none of these configuration variables can be selected by the user; they're all purely driven by the C library requirements.

The alternative would be to require that the C library implementations select *both* `COMMON_LIBC` along with the particular bit they need, and that seems redundant as well as error prone; if the C library selects only `COMMON_LIBC`, but not any particular piece, then you raise the issue from #56243. If you select only the piece and not `COMMON_LIBC`, then the library won't get built and the link will fail mysteriously.

Reply:
Technically, fixing #56243 should only require adding `zephyr_library_property(ALLOW_EMPTY TRUE)`; after all, there is nothing fundamentally wrong with having an empty "library," as long as that is one of the valid configurations.

Now, the question is: what is the purpose of the `COMMON_LIBC` Kconfig and why do we need it?

Since we seem to be making the "common libc" a standalone component (e.g. adding a dedicated doc page for it and exposing its configurations to the users via the common libc-specific Kconfigs like `COMMON_LIBC_MALLOC_ARENA_SIZE`), it would be quite natural to have a top level Kconfig (i.e. `COMMON_LIBC`) that sort of acts like a "master switch," in which case all subordinate Kconfigs should depend on the top level Kconfig (basically, the same idea as `MINIMAL_LIBC`, `NEWLIB_LIBC` and other "top level" libc Kconfigs").

On the other hand, if the `COMMON_LIBC` is only intended to be a helper symbol that indicates one or more common libc services are enabled, then the current implementation selecting `COMMON_LIBC` from the subordinate Kconfigs makes sense; but, this approach will likely require a refactoring in the future -- for instance, what if we decide to add an optional common libc service that is enabled by default (e.g. adding a prompt to `COMMON_LIBC_MALLOC` and making it `default y` such that the common libc malloc is used by default but the libc-specific malloc may also be used as fallback)?

Reply:
> Technically, fixing #56243 should only require adding `zephyr_library_property(ALLOW_EMPTY TRUE)`; after all, there is nothing fundamentally wrong with having an empty "library," as long as that is one of the valid configurations.
> 
> Now, the question is: what is the purpose of the `COMMON_LIBC` Kconfig and why do we need it?

Only because I was unaware of `zephyr_library_property(ALLOW_EMPTY_TRUE)`. We do not need it if we use that.

> On the other hand, if the `COMMON_LIBC` is only intended to be a helper symbol that indicates one or more common libc services are enabled, then the current implementation selecting `COMMON_LIBC` from the subordinate Kconfigs makes sense; but, this approach will likely require a refactoring in the future -- for instance, what if we decide to add an optional common libc service that is enabled by default (e.g. adding a prompt to `COMMON_LIBC_MALLOC` and making it `default y` such that the common libc malloc is used by default but the libc-specific malloc may also be used as fallback)?

Yup, I'll go remove `COMMON_LIBC` and add `zephyr_library_property(ALLOW_EMPTY_TRUE)` instead.

Reply:
done. This looks way better. Thanks!